proposals/lockout-violation-detection.md

```diff
@@ -0,0 +1,38 @@
+### Lockout Violation Detection
+
+We assume a database that we will store vote information ingested from gossip/turbine, which in turn will
+be consulted to detect lockout violations as follows.
+
+1) Track all SlotHashes on the main fork in the database.
+2) If there are any votes made that have a root `R` missing from the SlotHashes list, slash them on this fork
```

Comment:
What does slashing look like here? Forced destaking? Removing future rewards? Slashing the principal?

Reply:
I think full slashing could very much be highly centralizing for the chain. I think there should be a lot of discussion around it.

Reply:
> What does slashing look like here? Forced destaking? Removing future rewards? Slashing the principal?

It must be slashing principal sufficient to result in negative stake growth for the epoch, otherwise private self-staked validators can commit offenses with impugnity.

Reply:
Will be more discussion on the specific economics in a future proposal, this is simply designing methodology to detect.